src/libponyc/pass/syntax.c

```diff
@@ -909,6 +909,62 @@ static ast_result_t syntax_compile_error(pass_opt_t* opt, ast_t* ast)
   return AST_OK;
 }
 
+static ast_result_t syntax_lambda(pass_opt_t* opt, ast_t* ast)
+{
+  AST_GET_CHILDREN(ast, cap, name, typeparams, params, captures, type,
+    can_error, body);
+  switch(ast_id(type))
+  {
+    case TK_ISO:
+    case TK_TRN:
+    case TK_REF:
+    case TK_VAL:
+    case TK_BOX:
+    case TK_TAG:
+    {
+      errorframe_t info = NULL;
+      errorframe_t frame = NULL;
+      ast_error_frame(&frame, type, "lambda return type: %s",
+        ast_print_type(type));
+      ast_error_frame(&frame, type, "lambda return type cannot be "
+        "capability");
+      errorframe_append(&frame, &info);
+      errorframe_report(&frame, opt->check.errors);
```

Comment:
Note that you're not actually using the error frame feature here (`info` is never populated with anything, so appending it to `frame` has no effect).

So these 8 lines can be simplified to a call to `ast_error` and a call to `ast_error_continue`.

Here's an example you can copy from: https://github.com/ponylang/ponyc/blob/f019a19bc20c5064070a2a67c147dbd6480a0bcd/src/libponyc/pass/import.c#L57-L59

This will have the same effect in a more succinct way.